### DISCUSSION TITLE
Request for assistance with decrypting encrypted backups

### DISCUSSION THREAD

**Discussion by kotodan** [openssl/openssl#23568] [Q&A]
However, due to unforeseen circumstances, we urgently need to decrypt certain of our backup files today. We have attempted several decryption methods, but so far, we have been unable to access the encrypted data.
 We used the following command for encryption:
```
openssl smime -decrypt -in dhi.sql.bz2.ssl -binary \
 -inform DEM -inkey backup_key.pem -out dhi.sql.bz2
```
file size :2Gb

We would be extremely grateful if you could provide us with yourassistance, whether paid or unpaid, for decrypting these files as soon as possible. Your expertise in this field would be invaluable to us at this critical moment

**Comment by t8m** [comment]
This command is for decryption of S/MIME data. There is an error in there. `-inform DEM` should supposedly be `-inform DER`.

What happens if you use the (corrected) command?

**Reply by kotodan** [reply]
rror decrypting PKCS#7 structure
001E07055F7F0000:error:1C800064:Provider routines:ossl_cipher_unpadblock:bad decrypt:providers/implementations/ciphers/ciphercommon_block.c:124:

**Reply by kotodan** [reply]
Error decrypting PKCS#7 structure
001E07055F7F0000:error:1C800064:Provider routines:ossl_cipher_unpadblock:bad decrypt:providers/implementations/ciphers/ciphercommon_block.c:124:

**Reply by t8m** [reply]
This means either the data was somehow corrupted or the private key does not match the public key that was used for the encryption.

**Reply by kotodan** [reply]
The private key corresponds to the public key used for encryption. I tested encrypting a 3.5 GB file, but encountered the same error; it cannot be decrypted.

**Reply by t8m** [reply]
What is the openssl version used and what is the command used for the encryption?

**Reply by kotodan** [reply]
openssl 1.1.1f

**Reply by t8m** [reply]
Could you try later 1.1.1 version or even latest 3.2.1 version? There were some fixes in the S/MIME and CMS code related to handling binary files.
